### PR TITLE
KAFKA-6462: fix unstable ResetIntegrationTest

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @guozhangwang @bbejeck @dguy

**Review by guozhangwang — APPROVED** [review]

File: streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java [hunk 1]
```diff
@@ -167,11 +168,24 @@ public boolean conditionMet() {
     private final static Properties RESULT_CONSUMER_CONFIG = new Properties();
 
     void prepareTest() throws Exception {
-        cluster.deleteAndRecreateTopics(INPUT_TOPIC, OUTPUT_TOPIC, OUTPUT_TOPIC_2, OUTPUT_TOPIC_2_RERUN);
-
         prepareConfigs();
         prepareEnvironment();
 
+        // busy wait until cluster (ie, ConsumerGroupCoordinator) is available
+        while (true) {
```
Comment (dguy) [inline streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java:175]: Are we sure we want `while(true)`? Should we not give up after some period of time so the build doesn't hang?

Also, why can't we just put a longer wait in the `TestUtils.waitForCondition(...)`?

Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java:175]: The `while(true)` was in the code before the rewrite an not an issue.

A long wait won't do as `TestUtils.waitForCondition(...)` throws an `AsssertionException` if its wait time passed, while we get a `TimeoutException` from the request itself here if the broker is not available yet. I guess, we could increase the timeout for the request, but we reuse the same condition on other places and their we don't want to have  timeout -- could parametrize it of course....

Let me know what you think.

**Comment by brandonkirchner** [comment]
retest this please

**Comment by brandonkirchner** [comment]
whoops, meant to comment that on a different branch 😳

File: streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java (hunk 1 above)
Comment (dguy) [reply streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java:175]: no probs - leave as is then

**Comment by guozhangwang** [comment]
retest this please

**Pull request merged**